consultation_analyser/support_console/ingest.py

```diff
@@ -54,414 +44,478 @@ def get_all_question_part_subfolders(folder_name: str, bucket_name: str) -> list
     return question_folders
 
 
-def get_all_folder_names_within_folder(folder_name: str, bucket_name: str) -> set:
-    s3 = boto3.resource("s3")
-    objects = s3.Bucket(bucket_name).objects.filter(Prefix=folder_name)
-    set_object_names = {obj.key for obj in objects}
-    # Remove prefix
-    set_object_names = {name.replace("folder_name/", "") for name in set_object_names}
-    # Folders end in slash
-    folder_names = {name.split("/")[1] for name in set_object_names}
-    folder_names = set(folder_names) - {""}
-    return folder_names
-
-
-def get_folder_names_for_dropdown() -> list[dict]:
+def get_consultation_codes() -> list[dict]:
+    """
+    Get all available consultation codes from S3 for dropdown selection.
+    
+    Returns:
+        List of dicts with 'text' and 'value' keys for form dropdown
+    """
     try:
-        consultation_folder_names = get_all_folder_names_within_folder(
-            folder_name="app_data", bucket_name=settings.AWS_BUCKET_NAME
-        )
-    except RuntimeError:  # If no credentials for AWS
-        consultation_folder_names = set()
-    consultation_folders_formatted = [{"text": x, "value": x} for x in consultation_folder_names]
-    return consultation_folders_formatted
+        s3 = boto3.resource("s3")
+        objects = s3.Bucket(settings.AWS_BUCKET_NAME).objects.filter(Prefix="app_data/")
+        
+        # Get unique consultation folders
+        consultation_codes = set()
+        for obj in objects:
+            parts = obj.key.split("/")
+            if len(parts) >= 2 and parts[1]:  # Has consultation code
+                consultation_codes.add(parts[1])
+        
+        # Format for dropdown
+        return [{"text": code, "value": code} for code in sorted(consultation_codes)]
+    except Exception:
+        logger.exception("Failed to get consultation codes from S3")
+        return []
+
+
+def format_validation_error(error_message: str) -> dict[str, str]:
```

Comment:
This is a very long function - though it is quite hard to simplify.

I think 3 things could be abstracted out into separate functions which would make it a bit more readable:
`get_question_part(file_path)`
`get_file_name(file_path)`
`create_error_dictionary(type, message, technical, filepath=None)` - returning that dictionary.
(don't have to stick to those exact names).